ml-proto/README.md

```diff
@@ -5,10 +5,13 @@ This repository implements a prototypical reference interpreter for WebAssembly.
 Currently, it can
 
 * *parse* a simple S-expression format,
+* *decode* the binary format (work in progress),
 * *validate* modules defined in it,
-* *execute* invocations to functions exported by a module.
+* *execute* invocations to functions exported by a module,
+* *encode* the binary format,
+* *prettyprint* the S-expression format (work in progress).
 
-The file format is a (very dumb) form of *script* that cannot just define a module, but also batch a sequence of invocations.
+The S-expression format is a (very dumb) form of *script* that cannot just define a module, but also batch a sequence of invocations, assertions, and conversions. As such it is a superset of the binary format, with the additional functionality purely intended as testing infrastructure.
```

Comment:
Another useful thing to point out here is that .wast files can define a sequence of *modules* too, each with a sequence of tests following them.

Also, a lot of people (myself included) have been confused about .wast files. For me, they made sense when I realized that .wast files are not "module" files per se, or even containers for modules. If they define multiple modules, the modules don't interact; each module just replaces the previous one to be the "current" module -- semantics that would be surprising for any purpose other than testing. I'd suggest dropping the "superset" language, and saying something like this:

"As it processes its command-line arguments, it keeps track of a *current* module. When it encounters a module file, it decodes the module and assigns it as the current. When it encounters a test script file, it executes the test script. The `module` directive in the test script language creates a module and assigns it as current. Scripts may also contain invocations, assertions, and conversions, which perform tests on the current module."